packages/fxa-settings/src/components/UnitRowSecondaryEmail/index.tsx

```diff
@@ -2,55 +2,159 @@
     .map((email) => email.verified)
     .lastIndexOf(true);
 
-  // TODO: DRY this up and don't import `sentryMetrics` into every component
-  // needing a mutation - we must have an `onError` option in mutations
-  // to allow tests to pass but providing one prevents an error from actually
-  // throwing so we need to manually report it to Sentry.
-  // See https://github.com/apollographql/react-apollo/issues/2614
-  const [resendSecondaryEmailCode, { error }] = useMutation(
-    RESEND_SECONDARY_EMAIL_CODE_MUTATION,
-    {
-      onError: (error) => {
-        sentryMetrics.captureException(error);
-        revealAlertBar();
-      },
-      onCompleted: () => {
-        navigate('/beta/settings/emails/verify', { state: { email } });
-      },
-    }
+  const showSuccess = useCallback(
+    (message: string) => {
+      setAlertMessage([message, 'success']);
+      revealAlertBar();
+    },
+    [setAlertMessage, revealAlertBar]
+  );
+
+  const showError = useCallback(
+    (message: string) => {
+      setAlertMessage([message, 'error']);
+      revealAlertBar();
+    },
+    [setAlertMessage, revealAlertBar]
+  );
+
+  const [resendEmailCode] = useHandledMutation(RESEND_EMAIL_CODE_MUTATION, {
+    onCompleted() {
+      navigate('/beta/settings/emails/verify', { state: { email } });
+    },
+    onError(error) {
+      showError(`Sorry, there was a problem re-sending the verification code.`);
+      throw error;
+    },
+  });
+
+  const [makeEmailPrimary] = useHandledMutation(MAKE_EMAIL_PRIMARY_MUTATION, {
+    onCompleted() {
+      showSuccess(`${email} is now your primary email.`);
+    },
+    onError(error) {
+      showError(`Sorry, there was a problem changing your primary email.`);
+      throw error;
+    },
+    update: (cache) => {
+      cache.modify({
+        fields: {
+          account: (existing: Account) => {
+            const account = thaw(existing);
+            account.emails.find((m) => m.email === email)!.isPrimary = true;
+            account.emails.find(
+              (m) => m.isPrimary && m.email !== email
+            )!.isPrimary = false;
+            return account;
+          },
+        },
+      });
+    },
+  });
+
+  const [deleteEmail] = useHandledMutation(DELETE_EMAIL_MUTATION, {
+    onCompleted() {
+      showSuccess(`${email} email successfully deleted.`);
+    },
+    onError(error) {
+      showError(`Sorry, there was a problem deleting this email.`);
+      throw error;
+    },
+    update: (cache) => {
+      cache.modify({
+        fields: {
+          account: (existing: Account) => {
+            const account = thaw(existing);
+            account.emails.splice(
+              account.emails.findIndex((m) => m.email === email),
+              1
+            );
+            return account;
+          },
+        },
+      });
+    },
+  });
+
+  const SecondaryEmailUtilities = ({ children }: { children: ReactNode }) => (
+    <>
+      {verifiedAction && !verified && (
```

Comment:
@dannycoates Once [this](https://github.com/mozilla/fxa/commit/cc97bedbbe56bf92f8f2d77727b40cd2fec1d7fa#diff-ccd1a995f879047839c5065e2d190b29R68-R78) is merged I can remove these two conditions and feed everything straight in as props :)

Reply:
I forked off of this branch so feel free to just take it